### ISSUE TITLE
ProtonVPN 

### ISSUE THREAD

**Issue by GYKgamer** [Botspot/pi-apps#2617]
### What is the name of the app?

ProtonVPN

### Where is the app hosted?

https://protonvpn.com/download-linux

### About the app

ProtonVPN is simply, a vpn. Free to use, simple to install and works without box86/64.

### Upload file or Add PR Link

https://repo.protonvpn.com/debian/dists/stable/main/binary-all/protonvpn-stable-release_1.0.3-3_all.deb

### Confirmations

- [X] I have confirmed that this app is legal and not piracy.
- [X] I have confirmed that this app has never been discussed in https://github.com/Botspot/pi-apps/issues and https://github.com/Botspot/pi-apps/pulls, and it is not in the Pi-Apps app list.
- [X] I have read and followed the [creating an app for Pi-Apps documentation](https://pi-apps.io/wiki/development/Creating-an-app/).
- [X] I have confirmed that this app can run on all [Pi-Apps supported systems](https://pi-apps.io/install/).

**Comment by GYKgamer** [comment]
![image](https://github.com/user-attachments/assets/e73917c9-2cf8-46d9-8fc5-f200cae122a1)

**Comment by Botspot** [comment]
By any chance does this app add its own apt repository to the system?
We had an issue with this happening once, though in that case I'm not sure if the user added this repo or if the app added it automatically. https://github.com/Botspot/pi-apps/issues/959

**Comment by GYKgamer** [comment]
It does add it's own apt repo, after installing the deb you sue apt to install protonvpn

**Comment by theofficialgman** [comment]
the package linked is simply a package that contains the sources list and key. nothing else

![image](https://github.com/user-attachments/assets/8c15df6f-3f72-4c6e-b2b8-c9ff83ad02f6)

that is fine and pretty common for 3rd party repositories

all that is needed for pi-apps would be something like

```
# install latest repo package at the time of writing
install_packages https://repo.protonvpn.com/debian/dists/stable/main/binary-all/protonvpn-stable-release_1.0.3-3_all.deb || exit 1

# update repo package if newer version is available
apt_update
apt_lock_wait
sudo apt install --only-upgrade protonvpn-stable-release  -y

# install proton vpn
install_packages proton-vpn-gnome-desktop || exit 1
```

**Comment by theofficialgman** [comment]
the above could be made simpler if we obtained the latest version of the `protonvpn-stable-release` package by parsing the debian repo ourselves. similar to how this script does that https://github.com/Botspot/pi-apps/blob/master/.github/workflows/update_debian_repo_script.sh

**Comment by GYKgamer** [comment]
If possible, I would like to add more vpns? In #586 Botspot at the end added he wanted to add vpns to pi-apps, for example some others ones I use and like:
Windscribe VPN: https://github.com/Windscribe/Desktop-App
NordVPN: https://support.nordvpn.com/hc/en-us/articles/20196094470929-Installing-NordVPN-on-Linux-distributions

**Comment by Botspot** [comment]
Yes, I'm still supportive of adding VPNs. @GYKgamer, if you had the time, it would help us if you did more than add a link to the deb. Doing that just means that I put this on a mental list to install it myself, try it out, find a good text source to write the description, find the homepage to put in the website field, figure out how to uninstall it...
Most of those things would be easier for you to do, especially if you have prior experience with the app.

**Comment by GYKgamer** [comment]
I got a little something working for windscribe, it's just a starting point since I don't know how to determine a version number except for the one right one.
[windscribe.zip](https://github.com/user-attachments/files/16200661/windscribe.zip)

**Comment by Botspot** [comment]
@GYKgamer Were you using the pi-apps "create app" wizard for this? https://pi-apps.io/wiki/development/Creating-an-app/

**Comment by GYKgamer** [comment]
I read the documentation but I did not use the wizard

**Comment by Botspot** [comment]
I'm not sure where to begin. :(

- Need to use `install_packages`
- Need to use the script naming convention: `install-32`, `install-64`, `uninstall`, not `install32.sh` and `uninstall.sh`
- 64-bit packages should not ever be installed in `install-32`
- There should be `>/dev/null` after `sudo tee`
- `uninstall` should use `purge_packages`
- `version` variable should be used so it's easier to set up a github update script for the app

Please use the wizard and read the documentation. That will save everyone time.

**Comment by GYKgamer** [comment]
😆 maybe I wasn't as good at coding as I remembered, i'll use the wizard

**Comment by GYKgamer** [comment]
[Windscribe.zip](https://github.com/user-attachments/files/16202378/Windscribe.zip)
Updated, I used the wizard and followed the docs, hopefully should be better

**Comment by Botspot** [comment]
Better. Remaining issues:

- `purge_packages` takes no arguments. It wumatically uninstalls everything that was installed with `install_packages`.
- `install_packages` can take url arguments. So you can skip the `wget` and `rm` part by doing `install_packages "https://deploy.totallyacdn.com/desktop-apps/${version}/windscribe_${version}_arm64.deb"`

**Comment by GYKgamer** [comment]
[Windscribe.zip](https://github.com/user-attachments/files/16203423/Windscribe.zip)
that should be it now

**Comment by Botspot** [comment]
You are still trying to give arguments to `purge_packages`
What's with the `{echo ; exit 1}`? We have a function for this called `error`. Use that.

**Comment by GYKgamer** [comment]
My bad, I'm not as good at programming as I thought I was.
[Windscribe.zip](https://github.com/user-attachments/files/16203724/Windscribe.zip)

**Comment by Botspot** [comment]
@GYKgamer were you wanting to add Proton VPN or Windscribe, or both? IT seems that both are discussed in this issue.

**Comment by GYKgamer** [comment]
I kind of decided mid way through to change this into a discussion of vpns in general, If thats okay?
![image](https://github.com/user-attachments/assets/d05e3490-3053-46aa-9cc0-8cf07bd48b4e)

**Comment by Botspot** [comment]
@GYKgamer Were you still wanting to add Proton VPN?
What do you think about getting several vpns ready and then adding them to pi-apps all at once so I can put them in a new vpn category.

**Comment by GYKgamer** [comment]
> @GYKgamer Were you still wanting to add Proton VPN? What do you think about getting several vpns ready and then adding them to pi-apps all at once so I can put them in a new vpn category.

That was exactly the plan I had in mind.

**Comment by GYKgamer** [comment]
I wanted to bump this because of the current situation in the UK and how it could affect many people. It would definitely help with what's going on by adding VPNs.
